container/garray/garray_sorted_str.go

```diff
@@ -445,7 +445,7 @@ func (a *SortedStrArray) binSearch(value string, lock bool) (index int, result i
 	mid := 0
 	cmp := -2
 	for min <= max {
-		mid = (min + max) / 2
+		mid = min + int((max-min)/2)
```

Comment:
@lutherlau Thanks for your PR, but I do not see any bug here, would you please have a little bit explanation about this change?

Reply:
yeah, reference this [essay](https://ai.googleblog.com/2006/06/extra-extra-read-all-about-it-nearly.html)